### PR TITLE
Add support for container-overrides attribute

### PR DESCRIPTION
**Pull request by valaparthvi**
### What does this PR do?:

Adds support for container-overrides attribute, as described in https://github.com/devfile/api/issues/920#issuecomment-1244059075.

The logic for this PR is motivated by https://github.com/devfile/devworkspace-operator/pull/967.

### Which issue(s) this PR fixes:

Fixes part of https://github.com/devfile/api/issues/936

### PR acceptance criteria:
Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

> - Open new test/doc issues under the [devfile/api](https://github.com/devfile/api/issues) repo
> - Check each criteria if:
>  - There is a separate tracking issue. Add the issue link under the criteria
>  **or**
>  - test/doc updates are made as part of this PR
> -  If unchecked, explain why it's not needed

- [x] Unit/Functional tests

  

- [ ] [QE Integration test](https://github.com/devfile/integration-tests) 

  

- [ ] Documentation 

   
   Will work on docs once the PR is approved.

- [ ] Client Impact

  

- [ ] Gosec scans
  

### How to test changes / Special notes to the reviewer:
Override fields such as securityContext or resources of container by using `components[*].container.attributes` field.
Use the examples in https://github.com/devfile/api/issues/920#issue-1362465268.

### CODE REVIEW

File: pkg/devfile/generator/utils.go [hunk 1]
```diff
@@ -619,11 +625,87 @@ func getAllContainers(devfileObj parser.DevfileObj, options common.DevfileOption
...
+		}
+		if override.VolumeMounts != nil {
+			invalidFields = append(invalidFields, "volumeMounts")
+
+		}
+		if override.Env != nil {
+			invalidFields = append(invalidFields, "env")
+		}
+		if len(invalidFields) != 0 {
+			return fmt.Errorf("cannot use %s to override container %s", ContainerOverridesAttribute, strings.Join(invalidFields, ", "))
```
Comment (maysunfaisal) [inline pkg/devfile/generator/utils.go:678]:
```suggestion
			return fmt.Errorf("cannot use %s to override container due to missing %s", ContainerOverridesAttribute, strings.Join(invalidFields, ", "))
```

**Review by maysunfaisal — COMMENTED** [review]
small nit

**Review by maysunfaisal — APPROVED** [review]

**Comment by valaparthvi** [comment]
@maysunfaisal What do I do about coverage? Would you suggest I add unit tests covering them?

**Comment by maysunfaisal** [comment]
@valaparthvi yes that would be ideal and nice

File: pkg/devfile/generator/utils.go [hunk 2]
```diff
@@ -619,11 +623,87 @@ func getAllContainers(devfileObj parser.DevfileObj, options common.DevfileOption
...
 			}
 		}
-		containers = append(containers, *container)
+		// Check if there is an override attribute
+		if comp.Attributes.Exists(ContainerOverridesAttribute) {
+			patched, err := containerOverridesHandler(comp, container)
+			if err != nil {
+				return nil, err
+			}
+			containers = append(containers, *patched)
```
Comment (maysunfaisal) [inline pkg/devfile/generator/utils.go:632]: can we add a test case with attributes to cover this? I think we can add one to `TestGetContainers`

Comment (valaparthvi) [reply pkg/devfile/generator/utils.go:632]: Done: https://github.com/devfile/library/pull/157/commits/c0fab847058daa40af093bd0e7f6879c55e7fa67

**Review by maysunfaisal — APPROVED** [review]

**Pull request merged**